### ISSUE TITLE
Can't attach a container starting from succeed status

### ISSUE THREAD

**Issue by Crazykev** [hyperhq/hyperd#539]
How to reproduce:
```
$ sudo hyperctl run -d busybox sh
POD id is busybox-8211221991
Time to run a POD is 2428 ms
$ sudo hyperctl list container
Container ID                                                       Name                 POD ID               Status
4448eb62b4b50945a47cba27ac079f3c3798b439940d6147b6190b5a6755794d   busybox-8211221991   busybox-8211221991   running
$ sudo hyperctl stop -c 4448eb62b4b50
$ sudo hyperctl start -c 4448eb62b4b50
Successfully started container 4448eb62b4b50
$ sudo hyperctl attach 4448eb62b4b50
ls
ls
^C
```
This also apply in attach a container already attached and exited.

**Comment by gnawux** [comment]
Is this not fixed yet? @Crazykev

**Comment by Crazykev** [comment]
@gnawux I'm not working on this, and also need some change in runv.

**Comment by gnawux** [comment]
Thank you @Crazykev , I just confirm its status.

**Comment by gnawux** [comment]
confirm this bug, running but attach exit at once. 

daemon logs here:

```
I0403 05:22:18.641427  125105 server.go:152] Calling GET /v0.8.0/container/info
I0403 05:22:18.641459  125105 info.go:50] GetContainerInfo of busybox-3606458946
I0403 05:22:18.641531  125105 container.go:225] Pod[busybox-3606458946] Con[76e29e6ae62c(busybox-3606458946)] retrive info &types.ContainerStatus{Name:"busybox-3606458946", ContainerID:"76e29e6ae62c176b8d2e678692c67c4914af74d5a7b6644953b696e7d8b76904", Phase:"running", Waiting:(*types.WaitingStatus)(0xc42103eea0), Ru
nning:(*types.RunningStatus)(0xc42103eeb0), Terminated:(*types.TermStatus)(0xc420052370)} from status &pod.ContainerStatus{State:3, CreatedAt:time.Time{sec:63626808057, nsec:270934219, loc:(*time.Location)(0x1f5fd40)}, StartedAt:time.Time{sec:63626808118, nsec:889764424, loc:(*time.Location)(0x1f5fd40)}, FinishedAt:t
ime.Time{sec:63626808101, nsec:617087491, loc:(*time.Location)(0x1f144e0)}, ExitCode:0, Killed:false, RWMutex:sync.RWMutex{w:sync.Mutex{state:0, sema:0x0}, writerSem:0x0, readerSem:0x0, readerCount:1, readerWait:0}, stateChanged:(*sync.Cond)(0xc4210f7680)}
I0403 05:22:18.642486  125105 server.go:152] Calling POST /v1.17/attach
I0403 05:22:18.642573  125105 container.go:1287] Pod[busybox-3606458946] Con[76e29e6ae62c(busybox-3606458946)] attach: stdin: begin
I0403 05:22:18.642608  125105 container.go:1322] Pod[busybox-3606458946] Con[76e29e6ae62c(busybox-3606458946)] attach: stdout: begin
I0403 05:22:19.814805  125105 container.go:1313] Pod[busybox-3606458946] Con[76e29e6ae62c(busybox-3606458946)] attach: stdin: end
I0403 05:22:19.814849  125105 container.go:1336] Pod[busybox-3606458946] Con[76e29e6ae62c(busybox-3606458946)] attach: stdout: end
I0403 05:22:19.814862  125105 attach.go:30] Defer function for attach!
I0403 05:22:19.815054  125105 server.go:152] Calling GET /v0.8.0/container/info
I0403 05:22:19.815081  125105 info.go:50] GetContainerInfo of 76e29e6ae62c176b8d2e678692c67c4914af74d5a7b6644953b696e7d8b76904
I0403 05:22:19.815139  125105 container.go:225] Pod[busybox-3606458946] Con[76e29e6ae62c(busybox-3606458946)] retrive info &types.ContainerStatus{Name:"busybox-3606458946", ContainerID:"76e29e6ae62c176b8d2e678692c67c4914af74d5a7b6644953b696e7d8b76904", Phase:"running", Waiting:(*types.WaitingStatus)(0xc42103f570), Ru
nning:(*types.RunningStatus)(0xc42103f580), Terminated:(*types.TermStatus)(0xc420052550)} from status &pod.ContainerStatus{State:3, CreatedAt:time.Time{sec:63626808057, nsec:270934219, loc:(*time.Location)(0x1f5fd40)}, StartedAt:time.Time{sec:63626808118, nsec:889764424, loc:(*time.Location)(0x1f5fd40)}, FinishedAt:t
ime.Time{sec:63626808101, nsec:617087491, loc:(*time.Location)(0x1f144e0)}, ExitCode:0, Killed:false, RWMutex:sync.RWMutex{w:sync.Mutex{state:0, sema:0x0}, writerSem:0x0, readerSem:0x0, readerCount:1, readerWait:0}, stateChanged:(*sync.Cond)(0xc4210f7680)}
```

**Comment by gnawux** [comment]
Every std stream related issue is not simple, and this one is even more complicated than common cases. There are several issues in this case:

- container restart request will failed in hyperstart, which prevent restart an existing container ( hyperhq/hyperstart#287 )
- container start failure is ignored ( #584 )
- container restart won't re-connect std streams ( #585 )
- runv will reject streams reconnection because it did not cleanup streams status well ( hyperhq/runv#469 )

**Comment by gnawux** [comment]
finally it is fixed after hyperhq/hyperstart#306 closed
